.github/workflows/BuildPDFs.yml

```diff
@@ -13,16 +13,35 @@ jobs:
     steps:
       - name: Set up Git repository
         uses: actions/checkout@v4
-      - name: Compile LaTeX document
+      - name: Initialize and update submodules  #initialize and update submodules
+        run: |
+          git submodule init
+          git submodule update
+      - name: Compile LaTeX document    
         uses: xu-cheng/texlive-action@v2
         with:
           scheme: full
-          run: |
+          run: |            #runs build.py with submodule location and loopthrough examfolders and build pdf
             cd exercise
+            cp ../course_template/exerciseClass.cls ./
             python3 build.py
             cd ..
+
             cd lecture
+            cp ../course_template/lectureClass.cls ./
             python3 build.py
+            cd ..
+
+            for dir in exam/*/; do
+            if [ -f "$dir/build.py" ]; then
+              echo "Building in $dir"
+              cp ../../course_template/examClass "$dir"
+              cd "$dir" && python3 build.py && cd - 
+            else
+              echo "Skipping $dir: no build.py found."
+            fi
+            done
+            
```

Comment:
see other PR comment: this workflow will only work if we add (manually configured) build files per exam subfolder - I would appreciate a more automated routine

Reply:
Added a new build.py which will loop through the subdirectories and will automate the process of building pdfs,
Yaml file is changed accordingly .